src/common/io/src/format_settings.rs

```diff
@@ -13,145 +13,17 @@
-
-    // inner
-    pub nested: NestedFormatSettings,
-
-    // outer
-    pub true_bytes: Vec<u8>,
-    pub false_bytes: Vec<u8>,
-    pub null_bytes: Vec<u8>,
-    pub nan_bytes: Vec<u8>,
-    pub inf_bytes: Vec<u8>,
-    pub quote_char: u8,
-    pub escape: Option<u8>,
-
-    pub record_delimiter: Vec<u8>,
-    pub field_delimiter: Vec<u8>,
-    pub empty_as_default: bool,
-
-    pub json_quote_denormals: bool,
-    pub json_escape_forward_slashes: bool,
-    pub ident_case_sensitive: bool,
-
-    pub row_tag: Vec<u8>,
-}
-
-impl FormatSettings {
-    pub fn parse_escape(option: &str, default: Option<u8>) -> Option<u8> {
-        if option.is_empty() {
-            default
-        } else {
-            Some(option.as_bytes()[0])
-        }
-    }
-
-    pub fn parse_quote(option: &str) -> Result<u8> {
-        if option.len() != 1 {
-            Err(ErrorCode::InvalidArgument(
-                "quote_char can only contain one char",
-            ))
-        } else {
-            Ok(option.as_bytes()[0])
-        }
-    }
-
-    pub fn parse_row_tag(option: &str) -> Result<Vec<u8>> {
-        if option.is_empty() {
-            return Ok(vec![b'r', b'o', b'w']);
-        }
-        Ok(Vec::from(option))
-    }
-
-    pub fn for_values_parsing() -> Self {
-        Self {
-            timezone: "UTC".parse::<Tz>().unwrap(),
-            nested: Default::default(),
-
-            true_bytes: TRUE_BYTES_LOWER.as_bytes().to_vec(),
-            false_bytes: FALSE_BYTES_LOWER.as_bytes().to_vec(),
-            null_bytes: NULL_BYTES_UPPER.as_bytes().to_vec(),
-            nan_bytes: NAN_BYTES_LOWER.as_bytes().to_vec(),
-            inf_bytes: INF_BYTES_LOWER.as_bytes().to_vec(),
-            quote_char: b'\'',
-            escape: Some(b'\\'),
-
-            record_delimiter: vec![b'\n'],
-            field_delimiter: vec![b'\t'],
-
-            // not used
-            empty_as_default: true,
-            json_quote_denormals: false,
-            json_escape_forward_slashes: true,
-            ident_case_sensitive: false,
-            row_tag: vec![],
-        }
-    }
-
-    fn tsv_default() -> Self {
-        Self {
-            timezone: "UTC".parse::<Tz>().unwrap(),
-            nested: Default::default(),
-
-            true_bytes: TRUE_BYTES_NUM.as_bytes().to_vec(),
-            false_bytes: FALSE_BYTES_NUM.as_bytes().to_vec(),
-            nan_bytes: NAN_BYTES_LOWER.as_bytes().to_vec(),
-            inf_bytes: INF_BYTES_LOWER.as_bytes().to_vec(),
-            null_bytes: NULL_BYTES_ESCAPE.as_bytes().to_vec(),
-            quote_char: b'\'',
-            escape: Some(b'\\'),
-
-            record_delimiter: vec![b'\n'],
-            field_delimiter: vec![b'\t'],
-
-            // not used
-            empty_as_default: true,
-            json_quote_denormals: false,
-            json_escape_forward_slashes: true,
-            ident_case_sensitive: false,
-            row_tag: vec![],
-        }
-    }
 }
 
 // only used for tests
```

Comment:
Still only used for tests?

Reply:
used here https://github.com/datafuselabs/databend/issues/8949